lib/rubocop/cop/style/word_array.rb

```diff
@@ -53,13 +62,22 @@ def on_array(node)
 
         private
 
-        def complex_content?(strings)
+        def complex_content?(strings, word_regex: self.word_regex)
           strings.any? do |s|
+            next unless s.str_content
+
             string = s.str_content.dup.force_encoding(::Encoding::UTF_8)
-            !string.valid_encoding? || !word_regex.match?(string) || / /.match?(string)
+            !string.valid_encoding? ||
+              (word_regex && !word_regex.match?(string)) ||
+              / /.match?(string)
           end
         end
 
+        def invalid_percent_array_contents?(node)
+          # Disallow %w() arrays that contain invalid encoding or spaces
+          complex_content?(node.values, word_regex: false)
+        end
```

Comment:
Originally this used the same `complex_content?` check, taking the `WordRegex` configuration into account, however, this caused 206 offenses on rubocop itself, disallowing a lot of `%w()` arrays that are probably good to keep (ie. they contained non word characters). For instance: https://github.com/rubocop/rubocop/blob/281591717f37dab297358f695eb6e67c38352169/spec/rubocop/cop/lint/literal_as_condition_spec.rb#L4

As a result, I made the check only look for space issues. If desired we could add in another configuration to specify a regex for this check?